connection.go

```diff
@@ -0,0 +1,131 @@
+package papergres
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+// SSLMode defines all possible SSL options
+type SSLMode string
+
+const (
+	// SSLDisable - No SSL
+	SSLDisable SSLMode = "disable"
+	// SSLRequire - Always SSL, no verification
+	SSLRequire SSLMode = "require"
+	// SSLVerifyCA - Always SSL, verifies that certificate was signed by trusted CA
+	SSLVerifyCA SSLMode = "verify-ca"
+	// SSLVerifyFull - Always SSL, verifies that certificate was signed by trusted CA
+	// and server host name matches the one in the certificate
+	SSLVerifyFull SSLMode = "verify-full"
+)
+
+// Connection holds all database connection configuration.
+type Connection struct {
+	Database    string
+	User        string
+	Password    string
+	Host        string
+	Port        string
+	AppName     string
+	Timeout     int
+	SSLMode     SSLMode
+	SSLCert     string
+	SSLKey      string
+	SSLRootCert string
+}
+
+// NewConnection creates and returns the Connection object to the postgres server
+func NewConnection(databaseURL string, appName string) Connection {
+	u, err := url.Parse(databaseURL)
+	if err != nil {
+		panic(err)
+	}
+
+	host, port, _ := net.SplitHostPort(u.Host)
+	p, _ := u.User.Password()
+	q, _ := url.ParseQuery(u.RawQuery)
+	path := u.Path
+	if strings.Index(path, "/") == 0 {
+		path = path[1:]
+	}
+
+	// default
+	sslMode := SSLDisable
+	if len(q["sslmode"]) > 0 && q["sslmode"][0] != "" {
+		sslMode = SSLMode(q["sslmode"][0])
+	}
+
+	conn := Connection{
+		User:     u.User.Username(),
+		Password: p,
+		Host:     host,
+		Port:     port,
+		Database: path,
+		AppName:  appName,
+		SSLMode:  sslMode,
+	}
+
+	return conn
+}
+
+// String will build a connection string given database Connection settings
+func (conn *Connection) String() string {
```

Comment:
Yes, you're missing one more way to build connection strings.
There are two ways to make connection strings.
1. using Connection URIs
  The simplest way - like how you know shown above:
  ```
  postgresql://alpha:secret@localhost:port/testDB?param=value
  ```
2. using Data Source Name(DSN) strings
  DSN strings are made via using `keyword=value` pairs with space(s) between settings. DSN strings more commonly called connection strings. They offer more granular control, high extensibility and a wider range of parameters that can be configured for a DB and thus are preferred over connection URIs. Here's how it will look like:
```
host=localhost port=5432 dbname=testDB user=alpha password=secret connect_timeout=0

```
Read more about it here:
[Official postgres docs](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING) and [Go lib/pq driver](https://pkg.go.dev/github.com/lib/pq?utm_source=godoc#hdr-Connection_String_Parameters)

Reply:
Ok